libstuff/libstuff.cpp

```diff
@@ -2913,6 +2913,9 @@ void SRedactSensitiveValues(string& s) {
 
     // Remove anything inside "html" because we intentionally don't log chats.
     s = SREReplace("\"html\":\".*\"", s, "\"html\":\"<REDACTED>\"");
+
+    // Remove anything inside "edits" because these are also chats.
+    s = SREReplace("\"edits\":\\[\".*?\"\\]", s, "\"edits\":[\"REDACTED\"]");
```

Comment:
I coulddddd be wrong, BUT regex 101 is telling me:

<img width="439" alt="Screenshot 2024-09-24 at 8 38 02 PM" src="https://github.com/user-attachments/assets/32324459-3510-4f17-8baa-30bfb4148ea5">

The weird parts to me are:

1. `\\ matches the character \ with index 9210 (5C16 or 1348) literally (case sensitive)`
    - I don't think normally arrays in json look like `"edits":\...` right?
2. `Match a single character present in the list below [\".*?\"\\]`
    - I thinkkk you were going for trying to match the `[` character, then stuff inside of it right? It looks like this is doing the standard regex character grouping match thing, not just matching `[` (the start of an array) right?

Please tell me if I'm confused or a genius

Reply:
I see your confusion. I think it will click after I lay these points out for ya...

- We can't just pass `[*.]` as a regex string because brackets need to be escaped in regex if you are matching on them literally (otherwise they are special characters)
- a `\` character is _already_ an escape character as far as C++ strings go
- So, we have to escape twice.

That would get you something like this:

![2024-09-24_18-02-49](https://github.com/user-attachments/assets/5202d7ab-34ae-4266-9f5b-3509034a8891)

Maybe it would be better to use raw string literal? Then we would not need to double escape the `[` character only single escape and then it would be less weird or something?

Reply:
I think we can get rid of the quotes though and maybe simplify it to:

```c++
R"(\"edits\":\[.*?\])"
```

Reply:
That worked too so I pushed it.

Reply:
loveeeeeein that version, thanks for updating - looks much clearer to me!